src/main/scala/arithexpr/arithmetic/ArithExpr.scala

```diff
@@ -1783,14 +1784,14 @@ class Var private[arithmetic](val name: String,
   }
 
   override def visitAndRebuild(f: (ArithExpr) => ArithExpr): ArithExpr = {
-    f(Var(name, range.visitAndRebuild(f), Some(id)))
+    f(this.withRange(range.visitAndRebuild(f)))
   }
 
   override def substitute(subs: collection.Map[ArithExpr, ArithExpr]): Option[ArithExpr] = {
-    subs.get(this).orElse(range.substitute(subs).map(Var(name, _, Some(id))))
+    subs.get(this).orElse(range.substitute(subs).map(this.withRange))
   }
 
-  def copy(r: Range) = new Var(name, r, Some(this.id))
+  def copy(r: Range) = this.withRange(r)
```

Comment:
what is the difference between `withRange` and `copy`? are they redundant?

Reply:
Didn't stop to think about it in the hurry, but yes. Force pushed to just use `copy` for both `visitAndRebuild` and `substitute`